### PR TITLE
Fix for rapidly selecting and deleting a group of nodes in Script Canvas

### PR DESCRIPTION
**Pull request by spham-amzn**
Fixes an issue related to timing described in https://github.com/o3de/o3de/issues/7579 .

* When nodes are selected in Script Canvas, its components are created and displayed in the detail pane of the Script Canvas editor
* If you select a group of nodes, and then delete them AFTER the components are rendered in the node inspector, everything works as expected
* If you select the group of nodes but press the delete key very quickly, before the selected components are rendered (less than half a second), then a crash occurs because invalidating the windows checks against the component's entities but the component may have already been deleted

This does not address the root timing issue, but is a safe fix to protect against this race condition.

### CODE REVIEW

**Review by lsemp3d — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/Component/Component.cpp [hunk 1]
```diff
@@ -32,6 +33,7 @@ namespace AZ
     //=========================================================================
     Component::~Component()
     {
+        m_isDeleting = true;
```
Comment (amzn-sj) [inline Code/Framework/AzCore/AzCore/Component/Component.cpp:36]: Would it also work if we stored the `m_entity` in a temp and set it to `nullptr` before going into the if? Something like:
```
auto temp = m_entity;
m_entity = nullptr;
if (temp)
{
    temp->RemoveComponent(this);
}
```
Then we wouldn't need the bool and can keep the the logic for `GetEntityId()` and `GetNamedEntityId()`  the same.

**Review by amzn-sj — APPROVED** [review]

File: Code/Framework/AzCore/AzCore/Component/Component.cpp (hunk 1 above)
Comment (spham-amzn) [reply Code/Framework/AzCore/AzCore/Component/Component.cpp:36]: That would not completely prevent the crash I am seeing on Linux. If m_entity is set to null, then the original Component::GetEntityId() would continue to the following line:

```
        AZ_Warning("System", false, "Can't get component (type: %s, addr: %p) entity ID as it is not attached to an entity yet!", RTTI_GetTypeName(), this);
        return EntityId();
```
At this point, RTTI_GetTypeName() causes a segmentation fault.

**Comment by AMZN-alexpete** [comment]
Some kind of decision should be made/enforced on whether we want to allow any or partial access to AZ::Component from a separate thread because not every function is thread safe.    Perhaps the right thing to do in this case is to leave AZ::Component how it is, or assert if trying to access from another thread, and instead go to the Script Canvas UI and move that access to the main thread.

**Comment by spham-amzn** [comment]
> Some kind of decision should be made/enforced on whether we want to allow any or partial access to AZ::Component from a separate thread because not every function is thread safe. Perhaps the right thing to do in this case is to leave AZ::Component how it is, or assert if trying to access from another thread, and instead go to the Script Canvas UI and move that access to the main thread.

Yeah, that probably needs to happen, but would be more of an impactful change. This is just a safe bandage on the problem to prevent the crash for now, but I can create a GHI for the backlog

**Comment by AMZN-alexpete** [comment]
> Yeah, that probably needs to happen, but would be more of an impactful change. This is just a safe bandage on the problem to prevent the crash for now, but I can create a GHI for the backlog

I've created a GHI here https://github.com/o3de/o3de/issues/9637

**Pull request merged**